router/usecase/worker/candidate_route_search_data_worker.go

```diff
@@ -45,12 +64,44 @@ func (c *candidateRouteSearchDataWorker) OnPoolLiquidityCompute(ctx context.Cont
 	return nil
 }
 
-// ComputeSearchData implements domain.CandidateRouteSearchDataWorker.
-func (c *candidateRouteSearchDataWorker) ComputeSearchData(ctx context.Context, height uint64, blockPoolMetaData domain.BlockPoolMetadata) error {
-	// TODO: implement
-	// https://linear.app/osmosis/issue/DATA-248/[candidaterouteopt]-implement-and-test-core-pre-computation-logic
+func (c *candidateRouteSearchDataWorker) compute(blockPoolMetaData domain.BlockPoolMetadata) error {
```

Comment:
[CandidateRouteOpt] Unit test `compute` function in `candidate_route_search_data_worker.go`